src/main/java/core/basesyntax/service/StorageServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service;
+
+import core.basesyntax.controller.ControllerDao;
+import core.basesyntax.controller.ControllerDaoImpl;
+import core.basesyntax.model.CreateStrategy;
+import core.basesyntax.model.CreateStrategyImpl;
+import core.basesyntax.model.Fruit;
+import java.time.LocalDate;
+import java.util.List;
+
+public class StorageServiceImpl implements StorageService {
+    private ControllerDao<Fruit> controllerDao = new ControllerDaoImpl();
+    private CreateStrategy<Fruit> createStrategy = new CreateStrategyImpl();
+
+    public boolean supplyFruit(String fruitType,
+                               int count, LocalDate expirationDate) {
+        for (int i = 0; i < count; i++) {
+            if (!controllerDao.put(createStrategy.createFruit(fruitType, expirationDate))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean buyFruit(String fruitType,
+                            int count, LocalDate purchaseDate) {
+        List<Fruit> fruitList = controllerDao.getAll();
+        for (int i = 0; i < fruitList.size(); i++) {
+            if (count == 0) {
+                break;
+            }
```

Comment:
let's make this check before we call getAll() method, since if count is 0, we can exit method right away